toolkit/utilitynetworks/src/main/java/com/arcgismaps/toolkit/utilitynetworks/ui/TraceResultScreen.kt

```diff
@@ -0,0 +1,229 @@
+                            Column(horizontalAlignment = Alignment.End) {
+                                Text(
+                                    text = functionResult.functionType,
+                                    style = MaterialTheme.typography.titleSmall,
+                                    color = Color.Gray
+                                )
+                                Text(text = functionResult.value, style = MaterialTheme.typography.titleMedium)
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+}
+
+@Composable
+internal fun TraceResultSection(title: String, value: String, content: @Composable () -> Unit) {
+    Column(
+        modifier = Modifier
+            .fillMaxWidth()
+            .padding(top = 16.dp)
+    ) {
+        Text(
+            title,
+            color = Color.Gray,
+            style = MaterialTheme.typography.titleMedium,
+            modifier = Modifier.padding(start = 16.dp, bottom = 8.dp)
+        )
+        ExpandableCard(
+            initialExpandedState = false,
+            title = value,
+            padding = 0.dp,
+            modifier = Modifier.padding(horizontal = 16.dp)
+        ) {
+            content()
+        }
+    }
+}
+
+@Composable
+private fun ClearAllResultsButton(onClearAllResults: () -> Unit) {
+    Row(
+        horizontalArrangement = Arrangement.Center,
+        modifier = Modifier
+            .fillMaxWidth()
+            .padding(top = 16.dp)
+    ) {
+        Button(onClick = onClearAllResults) {
+            Text(stringResource(R.string.clear_all_results))
+        }
+    }
+}
+
+@Composable
+@Preview
+private fun TraceResultScreenPreview() {
+    val traceResults = TraceResults(
+        traceName = "Trace Name",
+        functionResults = listOf(
+            FunctionResult("Function 1", "1", "Add"),
+            FunctionResult("Function 2", " 2", "Average"),
+            FunctionResult("Function 3", " 3", "Count"),
+            FunctionResult("Function 1", "1", "Max"),
+            FunctionResult("Function 2", "2", "Min"),
+            FunctionResult("Function 3", "3", "Subtract"),
+            )
+    )
+    TraceResultScreen(traceResults, onClearAllResults = {}, onDeleteResult = {}, onZoomToResults = {})
+}
+/**
+ * Data class to hold the trace results.
+ *
+ * @param traceName The name of the trace.
+ * @param functionResults The list of function results.
+ * @since 200.6.0
+ */
+internal data class TraceResults(val traceName: String, val functionResults: List<FunctionResult>)
+
+/**
+ * Data class to hold the function result.
+ *
+ * @param name The name of the function result.
+ * @param value The value of the function result.
+ * @param functionType The type of the function result.
+ * @since 200.6.0
+ */
+internal data class FunctionResult(val name: String, val value: String, val functionType: String)
```

Comment:
to aid smart recomposition

```suggestion
@Immutable
internal data class FunctionResult(val name: String, val value: String, val functionType: String)
```